clients/src/main/java/org/apache/kafka/common/metrics/Metrics.java

```diff
@@ -419,6 +483,25 @@ public void run() {
         return Collections.unmodifiableMap(childrenSensors);
     }
 
+    public MetricName metricInstance(MetricNameTemplate template, String... keyValue) {
```

Comment:
I was wondering that myself, but didn't feel confident enough to make that suggestion.

I think there are still a couple places that I haven't yet converted over to using templates. I think there are some broker metrics that still use the existing API. I'm not at a computer right now, so I can't verify that.

The template is kind of like a more strict version of the metricName() API, that enforces documentation but also that predefines what keys you are allowed to put in the metric. It's less freeform than the current API. 

Is there any case where that we actually *want* a freeform API?